### PR TITLE
PA-16351 look for all contrib dev env vars

### PR DESCRIPTION
**Pull request by SOOS-JJennings**
### Changes:
  - Look for all contrib dev env vars across all ci/cd systems

**Ticket:** https://soos.atlassian.net/browse/PA-16351

### CODE REVIEW

File: src/services/AnalysisService.ts [hunk 1]
```diff
@@ -219,19 +390,34 @@ class AnalysisService {
 
     if (contributingDeveloperAudit?.length === 0) {
```
Comment (SOOS-GSteen) [inline src/services/AnalysisService.ts:391]:
```suggestion
    if (contributingDeveloperAudit === undefined) {
      contributingDeveloperAudit  = [];
    }

    if (contributingDeveloperAudit.length === 0) {
```

File: src/services/AnalysisService.ts [hunk 2]
```diff
@@ -190,6 +349,18 @@ class AnalysisService {
     }
   }
 
+  private findFirstEnvironmentVariableValue(environmentVariables: Array<string>): string | null {
+    for (const ev in environmentVariables) {
+      const environmentVariableValue = process.env[ev];
+
+      if (environmentVariableValue) {
```
Comment (SOOS-GSteen) [inline src/services/AnalysisService.ts:356]:
```suggestion
      if (environmentVariableValue && environmentVariableValue.length > 0) {
```

File: src/services/AnalysisService.ts [hunk 3]
```diff
@@ -219,19 +390,34 @@ class AnalysisService {
 
     if (contributingDeveloperAudit?.length === 0) {
       soosLogger.info(`Integration Name: ${integrationName}`);
-      const envVariableName = integrationNameToEnvVariable[integrationName];
-      if (envVariableName) {
-        const contributingDeveloper = process.env[envVariableName];
-        if (contributingDeveloper) {
+
+      // loop through all possible contributing developer environment variables and add the values for any that are found
+      for (const ev in contributingDeveloperEnvironmentVariables) {
```
Comment (SOOS-GSteen) [inline src/services/AnalysisService.ts:395]: you can add a bool param `stopAtFirst` defaulted to true to that func above, then set it to false for the call here, so you don't have to repeat the code - check the bool on the loop iteration

File: src/services/AnalysisService.ts [hunk 4]
```diff
@@ -219,19 +390,34 @@ class AnalysisService {
...
+      : this.findFirstEnvironmentVariableValue(commitHashEnvironmentVariables);
+    branchName = branchName
+      ? branchName
+      : this.findFirstEnvironmentVariableValue(branchNameEnvironmentVariables);
+    buildUri = branchUri
+      ? branchUri
+      : this.findFirstEnvironmentVariableValue(buildUriEnvironmentVariables);
+    buildVersion = buildVersion
+      ? buildVersion
+      : this.findFirstEnvironmentVariableValue(buildNumberEnvironmentVariables);
```
Comment (SOOS-GSteen) [inline src/services/AnalysisService.ts:419]: are these null or undefined? then you can do
```suggestion
    commitHash = commitHash ?? this.findFirstEnvironmentVariableValue(commitHashEnvironmentVariables);
    branchName = branchName ?? this.findFirstEnvironmentVariableValue(branchNameEnvironmentVariables);
    buildUri = branchUri ?? this.findFirstEnvironmentVariableValue(buildUriEnvironmentVariables);
    buildVersion = buildVersion ?? this.findFirstEnvironmentVariableValue(buildNumberEnvironmentVariables);
```

Comment (SOOS-JJennings) [reply src/services/AnalysisService.ts:419]: null, undefined or maybe ''

**Review by SOOS-GSteen — APPROVED** [review]

File: package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@soos-io/api-client",
-  "version": "1.0.14",
+  "version": "1.0.15",
```
Comment (SOOS-GSteen) [inline package.json:3]: redo lock

**Review by SOOS-GSteen — APPROVED** [review]

File: src/services/AnalysisService.ts [hunk 6]
```diff
@@ -217,19 +258,25 @@ class AnalysisService {
     soosLogger.info(`Creating scan for project '${projectName}'...`);
     soosLogger.info(`Branch Name: ${branchName}`);
 
-    if (contributingDeveloperAudit?.length === 0) {
+    if (contributingDeveloperAudit === undefined) {
```
Comment (SOOS-GSteen) [inline src/services/AnalysisService.ts:261]: do an assignment if using map 
```
if (contributingDeveloperAudit === undefined || contributingDeveloperAudit.length === 0) {
      soosLogger.info(`Integration Name: ${integrationName}`);
      contributingDeveloperAudit = contributingDeveloperEnvironmentVariables.map((ev) => {
        const environmentVariableValue = process.env[ev];
          return environmentVariableValue && environmentVariableValue.length > 0 ? {
            source: ContributingDeveloperSource.EnvironmentVariable,
            sourceName: ev,
            contributingDeveloperId: environmentVariableValue,
          } : null;
        }
      }).filter((a) => a !== null);
    }
```

**Review by SOOS-GSteen — APPROVED** [review]

**Pull request merged**